modules/local/dada2/main.nf

```diff
@@ -9,14 +9,24 @@ process DADA2 {
     tuple val(meta), path(reads)
 
     output:
-    tuple val(meta), path("*map.txt"), path("*asvs.fasta"), path("*_filt.fastq.gz"), emit: dada2_out
-    tuple val(meta), path("*_dada2_stats.tsv")                                     , emit: dada2_stats
+    tuple val(meta), path("*map.txt"), path("*asvs.fasta"), path("*_filt.fastq.gz"), optional: true, emit: dada2_out
+    tuple val(meta), path("*_dada2_stats.tsv")                                     , optional: true, emit: dada2_stats
+    tuple val(meta), env(stats_fail)                                               , optional: true, emit: dada2_stats_fail
     path "versions.yml"                                                            , emit: versions
     
     script:
     if ( meta.single_end ){
         """
-        dada2.R ${meta.id} $reads
+        set +e
+
+        error_file="dada2_errors.txt"
+        dada2.R ${meta.id} $reads 2> \$error_file
+
+        if [[ -s \$error_file ]] && grep -q "Error rates could not be estimated" \$error_file; then
+            stats_fail=true
+        else
+            stats_fail=false
+        fi
```

Comment:
I think we should try this error in the dada2.R file (a try catch.. not sure you can do that in R) and print our own message.

My thinking is that, the dada2 devs.. may change the error message in the future which would make this check fail. If we capture the error and print our own message.. we are in control.

Reply:
Also, I would do:
```
stats_fail=false
if [[ -s \$error_file ]] && grep -q "Error rates could not be estimated" \$error_file; then
   stats_fail=true
fi
```

Reply:
Also, if we can capture the error then we can exit with 0 and remove the `set +e`